src/Datastore/DatastoreClient.php

```diff
@@ -0,0 +1,796 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore;
+
+use DomainException;
+use Google\Cloud\ClientTrait;
+use Google\Cloud\Datastore\Connection\Rest;
+use Google\Cloud\Datastore\Query\GqlQuery;
+use Google\Cloud\Datastore\Query\Query;
+use Google\Cloud\Datastore\Query\QueryBuilder;
+use Google\Cloud\Datastore\Query\QueryInterface;
+use InvalidArgumentException;
+
+/**
+ * Google Cloud Datastore client. Cloud Datastore is a highly-scalable NoSQL
+ * database for your applications.  Find more information at
+ * [Google Cloud Datastore docs](https://cloud.google.com/datastore/docs/).
+ *
+ * Cloud Datastore supports [multi-tenant](https://cloud.google.com/datastore/docs/concepts/multitenancy) applications
+ * through use of data partitions. A partition ID can be supplied when creating an instance of Cloud Datastore, and will
+ * be used in all operations executed in that instance.
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder([
+ *     'projectId' => 'my-awesome-project'
+ * ]);
+ *
+ * $datastore = $cloud->datastore();
+ * ```
+ *
+ * ```
+ * // The Datastore client can also be instantianted directly.
+ * use Google\Cloud\Datastore\DatastoreClient;
+ *
+ * $datastore = new DatastoreClient([
+ *     'projectId' => 'my-awesome-project'
+ * ]);
+ * ```
+ *
+ * ```
+ * // Multi-tenant applications can supply a namespace ID.
+ * use Google\Cloud\ServiceBuilder;
+ *
+ * $cloud = new ServiceBuilder([
+ *     'projectId' => 'my-awesome-project'
+ * ]);
+ *
+ * $datastore = $cloud->datastore([
+ *     'namespaceId' => 'my-application-namespace'
```

Comment:
The settings provided here will override those set in the instantiation of the ServiceBuilder, so the projectId will also need to be provided.